src/screens/SearchScreen.js

```diff
@@ -14,12 +14,6 @@ import {map, navKeys} from '../constants';
 
 class SearchScreen extends Component {
     componentWillMount() {
-        const {token, navigation: {navigate, state: {key}}} = this.props;
-
-        if (!token && key !== navKeys.LOGIN) {
-            navigate(navKeys.LOGIN);
-        }
```

Comment:
The logic for navigation can be taken out since we're explicitly doing it with the actions now.